modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase

```diff
@@ -1201,6 +1201,155 @@ definition {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject162180");
+
+			ObjectPortlet.viewEntry(entry = "Entry Test");
+
+			ObjectPortlet.assertEntryColumnPresent(column = "External Reference Code");
+		}
+	}
+
+	@description = "LPS-158821 - Verify that the user is able to use the ERC field in Expression Builder, on the Validations tab"
+	@priority = "4"
+	test CanUseERCFieldWithExpressionBuilder {
+		task ("Create Custom Object and Custom Field (via API)") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 162179",
+				objectName = "CustomObject162179",
+				pluralLabelName = "Custom Objects 162179");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Integer",
+				fieldLabelName = "Custom Integer",
+				fieldName = "customInteger",
+				fieldType = "Integer",
+				isRequired = "false",
+				objectName = "CustomObject162179");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject162179");
+		}
+
+		task ("On Custom Object, go to validations tab") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 162179");
+
+			ObjectAdmin.goToValidationsTab();
+		}
+
+		task ("Add a Custom Validation using an expression 'is Greater Than'") {
+			ObjectCustomValidation.addObjectValidation(
+				validationName = "Custom Validation",
+				validationType = "Expression Builder");
+
+			ObjectCustomValidation.gotoCustomValidation(validationName = "Custom Validation");
+
+			ObjectCustomValidation.activeValidation();
+
+			ObjectCustomValidation.gotoTab(tabName = "Conditions");
+
+			ObjectCustomValidation.insertScriptOnValidation(scriptValidation = "customInteger > externalReferenceCode");
+
+			ObjectCustomValidation.addErrorMessage(errorMessage = "This entry is not possible.");
+
+			Button.clickSave();
+		}
+
+		task ("Create Custom View and go to it") {
+			ObjectCustomViews.goToViewsTab();
+
+			ObjectCustomViews.addObjectViewViaUI(viewName = "Custom Views");
+
+			ObjectCustomViews.goToViewsDetails(label = "Custom Views");
+		}
+
+		task ("Mark Custom View as default, go to View Builder and add ERC") {
+			ObjectCustomViews.markViewAsDefault();
+
+			ObjectCustomViews.goToViewsBuilderTab();
+
+			ObjectCustomViews.addColumnsViaUI(addColumns = "Custom Integer,External Reference Code");
+
+			Button.clickSave();
+		}
+
+		task ("Assert that Custom Validation with 'is Greater Than' is working") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject162179");
+
+			ObjectPortlet.addCustomFieldEntryViaUI(
+				customField = "Custom Integer",
+				entry = "123");
+
+			AssertTextEquals.assertPartialText(
+				locator1 = "Message#ERROR",
+				value1 = "This entry is not possible.");
+
+			Navigator.gotoBack();
+
+			ObjectPortlet.addCustomFieldEntryViaUI(
+				customField = "Custom Integer",
+				entry = "102030");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject162179");
+
+			ObjectPortlet.viewEntry(entry = "102030");
```

Comment:
@andreefarias what is the maximum possible value of the ERC? The integer value you use here should be greater than this.